bin/zend-expressive-swoole

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env php
+<?php
+/**
+ * Start, stop, and reload the HTTP server.
+ *
+ * @see       https://github.com/zendframework/zend-expressive-swoole for the canonical source repository
+ * @copyright Copyright (c) 2018 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-swoole/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Zend\Expressive\Swoole\Command;
+
+use PackageVersions\Versions;
+use Symfony\Component\Console\Application as CommandLine;
+use Symfony\Component\Console\CommandLoader\ContainerCommandLoader;
+
+// Setup/verify autoloading
+$cwd = getcwd();
```

Comment:
I tested this from weird `cwd`'s and wrote this to ascend up from __DIR__ and chdir when the root of the project was found:
```php
$depth = 1;
while ($depth <= 4) {
    $dir = sprintf('%s/%s', __DIR__, str_repeat('../', $depth));
    if (is_dir($dir . 'vendor')) {
        chdir($dir);
        $cwd = $dir;
        require 'vendor/autoload.php';
    }
    $depth++;
}

if (! isset($cwd)) {
    fwrite(STDERR, 'Cannot locate autoloader; please run "composer install"' . PHP_EOL);
    exit(1);
}
```

This would also make cwd correct for loading `pipeline.php` and `config.php` in the start command

Reply:
This approach does indeed work, in all scenarios except one that I tried: when symlinking the zend-expressive-swoole library into your vendor directory. In that scenario, it doesn't correctly traverse the tree to find the project root. However, that's quite an edge case (likely only people hacking on the library itself) and there are ways around it.

Thanks for the tip!